### DISCUSSION TITLE
Issue with Unique Constraint in Prisma Schema

### DISCUSSION THREAD

**Discussion by michaelvdveer** [prisma/prisma#22815] [Q&A] [answered]
### Question

I'm currently facing an issue with the unique: true constraint in my Prisma schema. I have a CommunityAdmin model with the following definition:
```prisma
model CommunityAdmin {
  id          String      @id @default(uuid())
  user        UserAccount @relation(fields: [userId], references: [id], onDelete: Cascade, onUpdate: Cascade)
  userId      String
  community   Community   @relation(fields: [communityId], references: [id], onDelete: Cascade, onUpdate: Cascade)
  communityId String
  deleted     Boolean     @default(false)
  deletedAt   DateTime?             
  createdAt   DateTime    @default(now())

  @@index([userId], name: "idx_community_admin_userId")
  @@index([communityId], name: "idx_community_admin_communityId")
  @@index([userId, communityId], name: "idx_community_admin_userId_communityId", unique: true)
  @@index([deleted])
}
```
Despite having unique: true on the composite index for userId and communityId, it seems not to enforce uniqueness as expected.

### How to reproduce (optional)

1. Define a model with a composite index and unique: true.
2. Attempt to insert duplicate records, expecting a uniqueness constraint violation.

### Expected behavior (optional)

**Expected Behavior:**

I expect the unique: true constraint on the composite index to enforce uniqueness for the specified fields.

**Actual Behavior:**

The uniqueness constraint does not seem to be enforced, and duplicate records can be inserted.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by michaelvdveer** [comment]
Additional info: Node version 18.18.0 / Prisma version 5.8.1 / Database: MySQL

**Comment by ludralph** [comment]
Hi @MichaelvdVeer 👋 

Thank you for raising this question.

Having `unique:true` at the composite index is not valid syntax for the `schema.prisma` file. 
This is the error I got below:

```
Error: P1012
error: No such argument.
  -->  schema.prisma:73
   | 
72 |   @@index([communityId], name: "idx_community_admin_communityId")
73 |   @@index([userId, communityId], name: "idx_community_admin_userId_communityId", unique:true)
```

Can you define your schema as shown below?
```prisma
model CommunityAdmin {
  id          String      @id @default(uuid())
  user        UserAccount @relation(fields: [userId], references: [id], onDelete: Cascade, onUpdate: Cascade)
  userId      String
  community   Community   @relation(fields: [communityId], references: [id], onDelete: Cascade, onUpdate: Cascade)
  communityId String
  deleted     Boolean     @default(false)
  deletedAt   DateTime?
  createdAt   DateTime    @default(now())

  @@index([userId], name: "idx_community_admin_userId")
  @@index([communityId], name: "idx_community_admin_communityId")
  @@index([userId, communityId], name: "idx_community_admin_userId_communityId")
  @@index([deleted])
}
```

**Comment by michaelvdveer** [comment]
Hi @ludralph  ! 
Thank you for you fast response.
I use the table CommunityAdmin to keep track of administrators of a community. I would like to ensure that a user can be an admin of multiple communities, but in the case of a specific community, the admin must be unique. Therefore, I deliberately added unique:true. However, it seems that this is not possible. Is there another way to achieve the same goal in the Prisma schema?

**Reply by ludralph** [reply, marked as answer]
Hi @MichaelvdVeer 

In Prisma, you can enforce [unique constraints](https://www.prisma.io/docs/orm/reference/prisma-schema-reference#unique) on single fields using the `@unique` attribute, or on [multiple fields](https://www.prisma.io/docs/orm/reference/prisma-schema-reference#unique-1) using the `@@unique` attribute. For example, to ensure that a combination of fields is unique, you would define it like this in your Prisma schema:
```prisma
model YourModel {
  // ... other fields ...

  @@unique([field1, field2])
}
```
In your case, you would want to do:
```prisma
@@unique([userId, communityId], name: "community_admin_userId_communityId")
```

**Reply by michaelvdveer** [reply]
Hi @ludralph 
Thank you this works!
